--- fbdesk-1.1.5/src/FbTk/FbWindow.cc.orig	2003-01-26 13:05:04.000000000 +0100
+++ fbdesk-1.1.5/src/FbTk/FbWindow.cc	2004-07-10 14:17:24.848613031 +0200
@@ -166,9 +166,12 @@
         return;
 
     Window root;
-    size_t border_width, depth;
+    unsigned int border_width, depth;
+    unsigned int tmp_width = m_width, tmp_height = m_height;
     XGetGeometry(s_display, m_window, &root, &m_x, &m_y,
-                 &m_width, &m_height, &border_width, &depth);
+                 &tmp_width, &tmp_height, &border_width, &depth);
+    m_width = tmp_width;
+    m_height = tmp_height;
 }
 
 void FbWindow::create(Window parent, int x, int y,
